modelservice/worker.py

```diff
@@ -72,12 +78,118 @@ def register(self):
+        else:
+            loss = "mean_absolute_error"
+
+        evaluation_metrics: list = []  # list(EvaluationMetric)
+
+        # Make models dir for job id
+        models_dir: str = f"{self.data_dir}/{request.id}"
+        os.mkdir(models_dir)
+
+        count = 1
+        gis_join_timer: Timer = Timer()
+        for gis_join in request.gis_joins:
+
+            gis_join_timer.start()
+            info(f"Loading data for GISJOIN {gis_join} ({count}/{len(request.gis_joins)})...")
+
+            # Load data
+            csv_path: str = f"{self.data_dir}/{gis_join}.csv"
+            all_df: pd.DataFrame = pd.read_csv(csv_path, header=0).drop("GISJOIN", 1)
+            len_df: int = len(all_df.index)
+            info(f"Loaded data for GISJOIN {gis_join}: {len_df} records")
+            if normalize_inputs:
+                scaled = MinMaxScaler(feature_range=(0, 1)).fit_transform(all_df)
+                all_df = pd.DataFrame(scaled, columns=all_df.columns)
+
+            features = all_df[feature_fields]
+            labels = all_df[label_field]
+
+            # Create Sequential model
+            model = tf.keras.Sequential()
+
+            # Add input layer
+            model.add(tf.keras.Input(shape=(len(request.feature_fields))))
+
+            # Add hidden layers
+            for hidden_layer in hyper_parameters.hidden_layers:
+                name: str = hidden_layer.name
+                units: int = hidden_layer.units
+                activation = "relu"
+                model.add(tf.keras.layers.Dense(units=units, activation=activation, name=name))
+
+            # Add output layer
+            output_layer: OutputLayer = hyper_parameters.output_layer
+            name: str = output_layer.name
+            activation = "relu"
+            model.add(tf.keras.layers.Dense(units=1, activation=activation, name=name))
+
+            # Compile the model and print its summary
+            model.compile(loss=loss, optimizer=optimizer)
+            model.summary()
+
+            # Fit the model to the data
+            history = model.fit(features, labels, epochs=epochs, validation_split=test_split)
+            hist = pd.DataFrame(history.history)
+            hist["epoch"] = history.epoch
+            info(hist)
+
+            last_row = hist.loc[hist["epoch"] == epochs - 1].values[0]
+            training_loss = last_row[0]
+            validation_loss = last_row[1]
+            info(f"Training loss: {training_loss}, validation loss: {validation_loss}")
+            gis_join_timer.stop()
+
+            metric: EvaluationMetric = EvaluationMetric(
+                training_loss=training_loss,
+                validation_loss=validation_loss,
+                duration_sec=gis_join_timer.elapsed,
+                error_occurred=False,
+                error_message="",
+                gis_join_metadata=GisJoinMetadata(
+                    gis_join=gis_join,
+                    count=len_df
+                )
+            )
```

Comment:
In our proposal, we also mentioned measuring benchmarks for disk I/O, network I/O, page fault rates, memory usage and CPU usage for the duration of each job. Do we want to include that in our metric object, as a separate metric or do we want to skip these benchmarks for this project?

Reply:
Can't get those from python runtime, has to be out of band